src/core/config/key_values.rs

```diff
@@ -27,15 +27,13 @@ pub struct KeyValue {
 }
 
 pub fn merge_key_value_vecs(current: &[KeyValue], other: &[KeyValue]) -> Vec<KeyValue> {
-    let mut acc: BTreeMap<&String, &String> =
-        current.iter().map(|kv| (&kv.key, &kv.value)).collect();
-
-    for kv in other {
-        acc.insert(&kv.key, &kv.value);
-    }
-
-    acc.iter()
-        .map(|(k, v)| KeyValue { key: k.to_string(), value: v.to_string() })
+    current
+        .iter()
+        .chain(other.iter())
+        .map(|KeyValue { key, value }| (key, value))
+        .collect::<BTreeMap<_, _>>()
+        .into_iter()
+        .map(|(k, v)| KeyValue { key: k.to_owned(), value: v.to_owned() })
         .collect()
 }
```

Comment:
There a little too much going on here. I think we can keep it simple and a bit more performant:

- Create one `HashMap<String, String>`
- Iterate over current and push all the values into the hashmap
- Next iterate over the second one and push the values into the hashmap
- Lastly convert the HashMap into Vec and return.

Your implementation doesn't basically the same thing, but it's not very clear from the implementation and does few unnecessary allocations on the way.

Reply:
Switched over!